shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/controllers/BridgeIngressController.java

```diff
@@ -163,26 +126,38 @@ public UpdateControl<BridgeIngress> reconcile(BridgeIngress bridgeIngress, Conte
             notifyManager(bridgeIngress, ManagedResourceStatus.READY);
             return UpdateControl.updateStatus(bridgeIngress);
         }
+
         return UpdateControl.noUpdate();
     }
 
     @Override
     public DeleteControl cleanup(BridgeIngress bridgeIngress, Context context) {
         LOGGER.info("Deleted BridgeIngress: '{}' in namespace '{}'", bridgeIngress.getMetadata().getName(), bridgeIngress.getMetadata().getNamespace());
 
-        // Linked resources are automatically deleted
+        // Linked resources are automatically deleted except for Authorization Policy and the ingress due to https://github.com/istio/istio/issues/37221
+
+        // Knative broker needs the dependent secret in order to be deleted properly https://coreos.slack.com/archives/CEXRYS5QC/p1649752951251439?thread_ts=1649752173.045369&cid=CEXRYS5QC
+        kubernetesClient.resources(KnativeBroker.class)
+                .inNamespace(bridgeIngress.getMetadata().getNamespace())
+                .withName(bridgeIngress.getMetadata().getName())
+                .delete();
+
+        // Since the authorizationPolicy has to be in the istio-system namespace due to https://github.com/istio/istio/issues/37221
+        // we can not set the owner reference. We have to delete the resource manually.
```

Comment:
Can you please create a ticket to track this? I would like to review it in the future